src/program/apis/tvmaze_api.py

```diff
@@ -0,0 +1,196 @@
+        response = self.request_handler.execute(
+            HttpMethod.GET,
+            url,
+            params={
+                "season": season,
+                "number": episode
+            }
+        )
+        
+        if not response.is_ok or not response.data:
+            return None
+
+        return self._parse_air_date(response.data)
+
+    def _parse_air_date(self, episode_data) -> Optional[datetime]:
+        """Parse episode air date from TVMaze response"""
+        if airstamp := getattr(episode_data, "airstamp", None):
+            try:
+                # Handle both 'Z' suffix and explicit timezone
+                timestamp = airstamp.replace('Z', '+00:00')
+                if '.' in timestamp:
+                    # Strip milliseconds but preserve timezone
+                    parts = timestamp.split('.')
+                    base = parts[0]
+                    tz = parts[1][parts[1].find('+'):]
+                    timestamp = base + tz if '+' in parts[1] else base + '+00:00'
+                elif not ('+' in timestamp or '-' in timestamp):
+                    # Add UTC timezone if none specified
+                    timestamp = timestamp + '+00:00'
+                # Convert to user's timezone
+                utc_dt = datetime.fromisoformat(timestamp)
+                return utc_dt.astimezone(self.local_tz)
+            except (ValueError, AttributeError) as e:
+                logger.debug(f"Failed to parse TVMaze airstamp: {airstamp} - {e}")
+
+        try:
+            if airdate := getattr(episode_data, "airdate", None):
+                if airtime := getattr(episode_data, "airtime", None):
+                    # Combine date and time with UTC timezone first
+                    dt_str = f"{airdate}T{airtime}+00:00"
+                    utc_dt = datetime.fromisoformat(dt_str)
+                    # Convert to user's timezone
+                    return utc_dt.astimezone(self.local_tz)
+                # If we only have a date, set time to midnight in user's timezone
+                local_midnight = datetime.fromisoformat(f"{airdate}T00:00:00").replace(tzinfo=self.local_tz)
+                return local_midnight
+        except (ValueError, AttributeError) as e:
+            logger.error(f"Failed to parse TVMaze air date/time: {airdate}/{getattr(episode_data, 'airtime', None)} - {e}")
+        
+        return None
+
+    def get_episode_release_time(self, item: MediaItem) -> Optional[datetime]:
+        """Get episode release time for a media item"""
+        if not isinstance(item, Episode):
+            logger.debug(f"Skipping release time lookup - not an episode: {item.log_string if item else None}")
+            return None
+            
+        if not item.parent or not item.parent.parent:
+            logger.debug(f"Skipping release time lookup - missing parent/show: {item.log_string}")
+            return None
+            
+        if not item.parent.parent.imdb_id:
```

Comment:
instead of all the `if not item.parent.parent` stuff, you can check the type by doing `if item.type == "episode"`

doing it like it is now can cause errors to popup when an item doesn't have the nested `parent` attributes